library/parser/json_reader.e

```diff
@@ -81,7 +81,8 @@ feature -- Commands
 	json_substring (start_index, end_index: INTEGER_32): STRING
 			-- JSON representation between `start_index' and `end_index'
 		do
-			Result := representation.substring (start_index, end_index)
+			-- Result := representation.substring (start_index, end_index)
+			create Result.make_from_string (representation.substring (start_index, end_index))
```

Comment:
I think this is useless to create a new STRING for Result ... as `substring` is already a new object.